### ISSUE TITLE
No IPv6 in System Information -> Network

### ISSUE THREAD

**Issue by treysis** [xbmc/xbmc#19493]
### Describe the bug
Here is a clear and concise description of what the problem is:

Looking at System Information -> Network, Kodi only shows the IPv4 address of the system, not the IPv6 addresses.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Kodi should show also IPv6 addresses, and not only IPv4.

### Your Environment
Used Operating system:

 - [x] Windows

 - Operating system version/name: Windows 10 Pro x64, Windows 11 Pro x64
 - Kodi version: 19/Matrix, 20/Nexus, 21/Omega

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
Just because I'm interested....Does Kodi actually show an IPv4 address at your setup or is that section simply empty and you just know that it usually shows an IPv4 at this section?

I'm asking because with current code, Kodi only shows the "first connected interface". So Kodi might probably show your IPv6 address if that's the only interface which is connected. If you have IPv4 *and* IPv6 configured, Kodi will search for the first interface and display whatever is found then. Think of the case where your LAN (ethernet) might be configured to use IPv6 and your WiFi interface might be configured with IPv4. Most OSs detect the WiFi adapter first

**Comment by treysis** [comment]
The device doesn't even have WiFi. All interfaces are dualstack. The correct IPv4 address is shown. There simply is no field to display the IPv6 address.

**Comment by DaVukovic** [comment]
Cool. Thanks. 

Then I would say Kodi behaves correctly due to the explanation I gave above. Kodi only shows the first connected interface. If it's the one with the IPv4 address, then Kodi does everything correct. 

Everything else would be an improvement for given functionally. 

Can you try to configure your system in a way that the interfaces do only have IPv6 addresses and can you then confirm the correct one is shown?

**Comment by treysis** [comment]
Well, "correctly" is subjective. All my interfaces are DualStack, so they have both IPv4 and IPv6 at the same time. I tried removing IPv4 from the interface, but it will still not show the IPv6 address. So yes, it's not a real bug (well, IPv6 support in 2021 should be better).

**Comment by DaVukovic** [comment]
Well, if no IPv4 is configured and IPv6 is not shown, then I would consider this being a bug as the address should be shown regardless of the version in use. 

I'll have a deeper look. Can't promise if I'm able to provide a fix, but we'll see ;)

**Comment by treysis** [comment]
Even if IPv4 is configured, it should still show IPv6, because both versions are used simultaneously.

**Comment by DaVukovic** [comment]
That would be the improvement and would be considered as a feature request. This section here is for bug reports. Feature requests should be submitted to our forums. 

It is a bug if only IPv6 is configured but not shown.
 It's not a bug if IPv6 is configured and also shown
It's not a bug if IPv4 and IPv6 is configured and only of them is being shown, as that's how current Kodi code is working (only showing the first connected interface). 

I guess explaining it for the 2nd time now is enough ;)

**Comment by treysis** [comment]
Yes, but it is only ONE interface. One interface can have multiple IP addresses.

**Comment by growse** [comment]
FWIW, on an IPv6-only host, kodi's Network page in the System Info shows "Link: Not connected" with the MAC address, IP, subnet etc. all blank.

> It is a bug if only IPv6 is configured but not shown.

Feels like this is true.

(19.3 installed on Debian buster from the raspbian repo, 20211118-a22c0ba3387)

**Comment by tiagogaspar8** [comment]
I notice this issue too, but in my case it shows the DNS server.

**Comment by treysis** [comment]
Still no change. Kodi doesn't seem to know about IPv6?

**Comment by treysis** [comment]
I was hoping this would get fixed with 20/Nexus, but nothing has changed :(

**Comment by lrusak** [comment]
> I was hoping this would get fixed with 20/Nexus, but nothing has changed :(

PR's are welcome to help improve the status page.

**Comment by stuaxo** [comment]
Hi @github-actions bot please un-stale this, mixed IPV4 IPV6 networks are still a thing.

**Comment by tiagogaspar8** [comment]
This is still valid.

**Comment by treysis** [comment]
Updated the bug report to reflect that this issue also affects Kodi 21/Omega.

**Comment by bjsvedin** [comment]
I had the same thought as the others did. For anyone curious I have a [WIP branch](https://github.com/bjsvedin/xbmc/tree/ipv6) that adds displaying BOTH IPv4 and IPv6 address on the System Info pages. I would like feed back on what others want from this. My thoughts were to add the V6 address, Gateway, and Nameserver alongside the V4 variants.

I have tested that branch ONLY on Linux so far. I will get to Windows at some point, and maybe Android. Mac and iOS will be problematic. Until the others are tested those implementations are made blind. I have no idea if they even build.

One limitation I have run into and am not sure how to solve is V6 Temporary addresses. From my research, there is no way to guarantee the first address pulled is the stable address, nor is there any indication as to which one is, though it will likely be the first by coincidence. Any thoughts on this issue is welcome.

**Comment by garbear** [comment]
I don't see how the IPv6 commit could hurt, even if it doesn't show the best IPv6 commit at this point at least it's more informative than purely IPv4 display. In regards to the temporary address, is there a canonical way to determine this? If not, maybe show every IPv6 address, or accept that wrong info is better than no info.

**Comment by treysis** [comment]
Yes, with IPv6 you should show ALL addresses. However, the OS will always prefer the temporary one. I believe there is a way to determine this but I'd have to look into it.

Thx for the effort, though. I always had this on my bucket list but...you know, tempus fugit!

I agree that _some_ IPv6 address is better than _no_ IPv6 address!

**Comment by bjsvedin** [comment]
After some further research into this topic, and how that window in kodi is setup I have a plan. 

I think I can get the first Stable address by moving the V6 logic into each platform specific implementation of CNetworkInterface. My first proof of concept left it in the posix implementation and it's not possible there. 

I don't think i can display ALL v6 addresses due to the currently limits in the SystemInfo xml, it has a limit of twelve entries. I don't want to make this update too large, and start changes the xml and window layouts. I also don't know how the xml and other skins work, so I don't want to dig myself too deep too fast.

That leaves 4 extra slots I can fill. That being the stable V6 address, V6 Gateway, V6 Name Server, and the last being either a 2nd name server, or maybe the link local. If anyone has a preference for that 4th one let me know.

**Comment by treysis** [comment]
Can't you make the field in the XML just represent a long string, e.g. a comma-separated list of all v6 addresses but handled as a single string?
Same for DNSv6.

But sure, that should be also quite easy to add later on.

**Comment by bjsvedin** [comment]
I tested the limits of one of the rows and repeated the ip address multiple times with many new lines. It has hard limits to what it will display. Two wide, and two deep. Anything more in either direction does not get rendered. As well it starts to push onto other entries.

I did briefly consider repurposing the xml item used for the privacy policy and manually rendering out the entire network page. This would allow for scrolling and a full list of IP addresses, name servers, and gateways. Possibly even across multiple interfaces and their network speeds. At that point we could do as much as we wanted. I could revisit this idea since the xml element for it already exists it won't be that large of a change.

My initial plan was to not reshape in its entirety the network display page, but just to add some  V6 basics.

<img width="1056" height="107" alt="Image" src="https://github.com/user-attachments/assets/095c35a5-f08c-4588-ac9a-633171fd2f87" />
<img width="1884" height="185" alt="Image" src="https://github.com/user-attachments/assets/9c55f33e-b19e-4e31-92be-8bc98ed2ef24" />

**Comment by garbear** [comment]
Here's another idea, you have 12 individual line items, 1 massive text blob, and you're trying to shoehorn both into a new novel use case. Don't be afraid to do something completely new that fits the situation better than 12 text lines or 1 text blob.

**Comment by treysis** [comment]
I like either way. I'd prefer the fast, limited version though. Gets IPv6 out there faster :)

**Comment by bjsvedin** [comment]
@garbear  I don't want to modify XML for this initial update. Just getting the basics is my initial plan. I can revisit this with a unique solution to viewing all the data afterwards.

**Comment by garbear** [comment]
Agreed, a two-step process here is best. Step 1, make it work. Step 2, make it work good.

**Comment by bjsvedin** [comment]
Here is the basic update results. I have only implemented Linux so far. But with one platform done as a template, I expect the rest to go smoothly.

<img width="2537" height="1373" alt="Image" src="https://github.com/user-attachments/assets/7c16eb9d-8b5f-4987-a725-ecdefe124672" />

**Comment by treysis** [comment]
Very cool! Will you create a PR?

**Comment by bjsvedin** [comment]
Sure, here is a draft PR. https://github.com/xbmc/xbmc/pull/28296

**Comment by bjsvedin** [comment]
I have added an Android implementation.

**Comment by garbear** [comment]
Nice! Sorry no one has reviewed the PR, it's kinda like whenever devs have free time, but it's all good code, so hopefully it gets some eyes soon.

**Comment by bjsvedin** [comment]
Oh I am not expecting a review yet. It's still a draft, and I am still working on implementing for all the platforms that I can. I won't turn down any feed back that comes around though.

**Comment by malard** [comment]
*This reply was written and posted by Claude on @malard's behalf.*

Flagging a status change nobody has noted here: @bjsvedin's draft, xbmc/xbmc#28296, was **closed unmerged on 2026-08-16**. This report currently has no fix in flight, which is not obvious from reading the thread - the last comments here are about the PR progressing well.

@bjsvedin - if you closed it because it stalled waiting for review rather than because the approach was wrong, please say so; the work covered several platforms and is worth reopening rather than restarting.

Labelling `Component: Network` so it is findable - this issue has carried no labels at all for five years.

**Comment by treysis** [comment]
@malard I also couldn't find the suggested "bad non-draft PR"...

**Comment by stuaxo** [comment]
@treysis you might be able to cherry pick the commits https://github.com/xbmc/xbmc/pull/28296/commits if you want to try it.

**Comment by bjsvedin** [comment]
Sorry, i haven't gotten around to finishing up this feature, i ran out of time at the end of it. At one point I had it building on all the platforms, but the last time i tried it didn't compile on windows and i don't know why. I may get back to this at some point, or some else is welcome to pick up where i left off.
